src/ui/settingsPage/tableElement.js

```diff
@@ -0,0 +1,295 @@
+      background-position: center center;
+      background-repeat: no-repeat;
+      margin-inline-start: 8px;
+      pointer-events: none;
+      transform: rotate(180deg);
+      transition: transform 0.275s ease-in-out;
+      inline-size: 24px;
+      height: 24px;
+    }
+    .tableHolder td {
+      padding: 16px;
+    }
+    .deleteBtn {
+      background-color: transparent;
+      color: transparent;
+      border: transparent;
+      background-image: url("../../assets/img/delete-gray.svg");
+      background-position: center center;
+      background-repeat: no-repeat;
+      transition: ease-in-out 0.3s;
+    }
+    .deleteBtn:hover {
+      filter: brightness(1.5);
+    }
+    td.delete {
+      width: 30px;
+    }
+    .row {
+      display: flex;
+      align-content: center;
+      align-items: center;
+    }
+  `;
+}
+customElements.define("mz-context-table", ContextTable);
+
+export const tableHeading = (
+  sortedBy = "",
+  acending = true,
+  sortBy = () => {}
+) => {
+  const getClass = (id) => {
+    if (sortedBy != id) {
+      return "";
+    }
+    if (acending) {
+      return "sorted-down";
+    }
+    return "sorted-up";
+  };
+
+  const onClick = (clickedId) => {
+    // If it's the same, just toggle sort type.
+    if (clickedId === sortedBy) {
+      sortBy(clickedId, !acending);
+      return;
+    }
+    // Otherwise just asc search.
+    sortBy(clickedId, true);
+  };
+
+  return html`
+    <tr class="tableHeader">
+      <th>
+        <button class=${getClass("origin")} @click=${() => onClick("origin")}>
+          Website
+        </button>
+      </th>
+      <th>
+        <button
+          class=${getClass("excluded")}
+          @click=${() => onClick("excluded")}
+        >
+          VPN status
+        </button>
+      </th>
+      <th>
+        <button class=${getClass("city")} @click=${() => onClick("city")}>
+          Location
+        </button>
+      </th>
+      <th></th>
+    </tr>
+  `;
+};
+
+export const noElementPlaceHolder = (contexts = []) => {
+  if (contexts.length != 0) {
+    return null;
+  }
+  return html`
+    <div class="emptyState">
+      <img src="../../assets/img/country-tabs.svg" />
+      <h3>You havent set any website preferences yet</h3>
+      <p>
+        If you turn off VPN protection or set a custom location for any
+        websites,you'll see them here.
+      </p>
+    </div>
+  `;
+};
+
+/**
+ * @param {SiteContext} ctx
+ */
+export const tableRow = (ctx, serverList, removeOrigin) => {
+  const name = ctx.excluded
+    ? ""
+    : Utils.nameFor(ctx.countryCode, ctx.cityCode, serverList);
+
+  return html`
+    <tr class="">
+      <td>
+        <a href="https://${ctx.origin}">${ctx.origin}</a>
+      </td>
+      <td class="row">
+        <img src="../../assets/img/shield-${ctx.excluded ? "off" : "on"}.svg" />
+        ${ctx.excluded ? "OFF" : "ON"}
+      </td>
+      <td>${name}</td>
+      <td class="delete">
+        <button
```

Comment:
Needs aria label and tooltip

Reply:
Dont think buttons need a label tho? In the accessibility tools this seems to be fine.

Reply:
you're right, my bad! I missed the 'remove' text on L290.